tools/bin/check_requirements.sh

```diff
@@ -1,68 +1,102 @@
+#!/usr/bin/env bash
+
+# Ensure always run from this directory because it uses relative paths
+cd "$(dirname "${0}")" || exit 1
+
+echo "Checking Airbyte development tools and versions.."
+
 printf "Docker ";
+# Currently no pinned docker version: https://github.com/airbytehq/airbyte/issues/9229
 if [[ $(which docker) && $(docker --version) ]]; then
     printf "is installed"
   else
     printf "needs to be installed"
 fi;
+
 printf "\n";
-desired="14"
+DESIRED_JAVA_VERSION="$(cat ../../.java-version)"
 printf "Java ";
-if [[ "$(which java)" && "$(java --version)" ]]; 
+if [[ "$(which java)" && "$(java --version)" ]];
     then
-        printf "installed"
         str="$(java --version)"
+        # Returns like "openjdk 11.0.12 2021-11-24"
         IFS=' ' read -ra array <<< "${str}"
-        version="${array[1]}"
-        if [[ "${version}" > "${desired}" || "${version}" == "${desired}" ]];
+        java_version="${array[1]}"
+        printf "${java_version} installed"
+        if [[ "${java_version}" == "${DESIRED_JAVA_VERSION}" ]];
             then
-                printf " and functional"
+                printf " and matches version ${DESIRED_JAVA_VERSION}"
             else
-                printf " but not functional, must have version ${desired} at least"
+                printf " but does not match version ${DESIRED_JAVA_VERSION}, you might see unexpected behavior"
```

Comment:
Changed the logic here to require the exact version. Using a higher version could have issues and it should not be "okay" if someone has a higher version than the required version even if it is better than having a lower version.